### ISSUE TITLE
Support Gson's `@Expose` annotation

### ISSUE THREAD

**Issue by robfletcher** [robfletcher/grails-gson#12]

**Comment by beckje01** [comment]
Would you like a different converter that supports the `@Expose` or do you want the current converter to be driven by a config option?

**Comment by robfletcher** [comment]
Good question. I hadn't really given it much thought. It doesn't feel like much would need to be different other than the mechanism for selecting the properties to iterate over. There are a couple of side-questions such as what happens if you add `@Expose` to a transient non-persistent property?

**Comment by beckje01** [comment]
The only reason you might want a different converter is if you wanted to be able to both do a as GSON that does the whole object as well as a as ExposedGSON that only shows the items with `@Expose` 

As far as transient non-persistent property I haven't really tested with that in mind could it even be the same annotation?

**Comment by robfletcher** [comment]
Yeah, a different converter might make sense if you wanted to handle different classes with one or other mechanism. I think creating a different `JsonSerializer` / `JsonDeserializer` class might make sense. Another option would be refactoring the existing code to use an exclusion strategy.

**Comment by beckje01** [comment]
Just a quick idea what do you think of this I can put together a pull request if you like it https://gist.github.com/beckje01/5145266

**Comment by jrgleason** [comment]
I just noticed @Exclude wasn't working for me. Do the other annotations function as expected? Will this ever be resolved?
